ir/src/lib.rs

```diff
@@ -193,6 +213,30 @@ mod tests {
         assert!(result.is_err());
     }
 
+    #[test]
+    fn err_bc_empty_or_omitted() {
+        // if boundary constraints are empty, an error should be returned at parser level.
+        let source = "
+        trace_columns:
+            main: [clk]
+        boundary_constraints:
+        transition_constraints:
+            enf clk' = clk + 1";
+
+        assert!(parse(source).is_err());
+
+        // if boundary constraints are omitted, an error should be returned at IR level.
+        let source = "
+        trace_columns:
+            main: [clk]
+        transition_constraints:
+            enf clk' = clk + 1";
+
+        let parsed = parse(source).expect("Parsing failed");
+        let result = AirIR::from_source(&parsed);
+        assert!(result.is_err());
+    }
+
```

Comment:
Public inputs are missing here and in some of the other tests, so that may affect these tests. In the future we'll want to test exact errors like we do in Miden VM, but that can wait until we've revised our unit testing as part of v0.2

Reply:
Fixed, thanks.